ui/src/AppCharting/chart-data-extractor.ts

```diff
@@ -134,20 +153,42 @@ export async function getChartDataByTimestamp(
     const column: ChartingArray = [dataName];
 
     for (const timestamp of timestamps) {
-      let value = 0;
+      let value: number = MISSING_VALUE;
       const benchmark = benchmarkMap[timestamp.key];
-      const scenarioSizeRecords = benchmark && benchmark[scenario];
-      if (benchmark && scenarioSizeRecords) {
-        const plotTypeRecords =
-          scenarioSizeRecords[isFan ? `${dataName}_${serverCount}` : dataName];
-        if (plotTypeRecords) {
-          const plotStatisctics = plotTypeRecords[plotType];
-          if (plotStatisctics) value = plotStatisctics[statName];
+      if (benchmark) {
```

Comment:
oh man... there's probably some special handling here but it sure looks a whole lot like the handler above.

Reply:
I agree, there was already some duplication in the handlers and I probably added to it a bit. But sometime soon there will be an additional follow-on task to make more extensive use of the new parameter metadata. Those changes will likely require more extensive changes to the functions that extract data from the benchmark map, so I think it's OK to leave it for now and I'll make a note to try to reduce duplication as part of the next task.